src/main/java/uk/gov/hmcts/cmc/claimstore/documents/BulkPrintService.java

```diff
@@ -53,30 +51,38 @@ public BulkPrintService(
         this.appInsights = appInsights;
     }
 
-    @EventListener
     @LogExecutionTime
     @Retryable(
         value = {HttpClientErrorException.class, HttpServerErrorException.class},
         backoff = @Backoff(delay = 200)
     )
-    public void print(DocumentReadyToPrintEvent event) {
+    public void print(Claim claim, Document defendantLetterDocument, Document sealedClaimDocument) {
+        requireNonNull(claim);
+        requireNonNull(defendantLetterDocument);
+        requireNonNull(sealedClaimDocument);
+
         sendLetterApi.sendLetter(
             authTokenGenerator.generate(),
-            new Letter(
-                Arrays.asList(event.getDefendantLetterDocument(), event.getSealedClaimDocument()),
-                XEROX_TYPE_PARAMETER, wrapInMap(event.getClaim())
+            new Letter(Arrays.asList(defendantLetterDocument, sealedClaimDocument),
+                XEROX_TYPE_PARAMETER, wrapInMap(claim)
             )
         );
     }
 
     @Recover
-    public void notifyStaffForBulkPrintFailure(DocumentReadyToPrintEvent event) {
+    public void notifyStaffForBulkPrintFailure(
+        RuntimeException exception,
+        Claim claim,
+        Document defendantLetterDocument,
+        Document sealedClaimDocument
+    ) {
         bulkPrintStaffNotificationService.notifyFailedBulkPrint(
-            event.getDefendantLetterDocument(),
-            event.getSealedClaimDocument(),
-            event.getClaim()
+            defendantLetterDocument,
+            sealedClaimDocument,
+            claim
         );
-        appInsights.trackEvent(BULK_PRINT_FAILED, "referenceNumber", event.getClaim().getReferenceNumber());
+
+        appInsights.trackEvent(BULK_PRINT_FAILED, "referenceNumber", claim.getReferenceNumber());
```

Comment:
I know this is existing code, but would be nice to use `uk.gov.hmcts.cmc.claimstore.appinsights.AppInsights#REFERENCE_NUMBER` constant instead of 
 `String` "referenceNumber" here.